OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -331,7 +340,9 @@ void SpawnPassenger(Actor passenger)
 				w.Add(passenger);
 				passenger.Trait<IPositionable>().SetPosition(passenger, self.Location);
 
-				// TODO: this won't work well for >1 actor as they should move towards the next enterable (sub) cell instead
+				var mobile = passenger.TraitOrDefault<Mobile>();
```

Comment:
Can you use `OnNotifyBlockingMove` instead?  Hardcoding mobile references through unrelated code kind of sucks.